backend/tests/v5/e2e/routes/sso/aad/aad.test.js

```diff
@@ -0,0 +1,180 @@
+			const res = await agent.post('/v5/sso/aad/signup').send({ ...newUserData, username: testUser.user })
+				.expect(templates.invalidArguments.status);
+			expect(res.body.code).toEqual(templates.invalidArguments.code);
+		});
+
+		test('should fail if there are missing body params', async () => {
+			const res = await agent.post('/v5/sso/aad/signup').send({ ...newUserData, username: undefined })
+				.expect(templates.invalidArguments.status);
+			expect(res.body.code).toEqual(templates.invalidArguments.code);
+		});
+
+		test('should validate signup data and redirect the user to Microsoft authentication page', async () => {
+			const res = await agent.post('/v5/sso/aad/signup').send(newUserData)
+				.expect(templates.found.status);
+
+			const redirectUri = res.headers.location;
+			expect(redirectUri).toEqual(
+				expect.stringContaining('https://login.microsoftonline.com/common/oauth2/v2.0/authorize'),
+			);
+			expect(redirectUri).toEqual(
+				expect.stringContaining(`redirect_uri=${encodeURIComponent(signupRedirectUri)}`),
+			);
+			expect(redirectUri).toEqual(
+				expect.stringContaining(`state=${encodeURIComponent(JSON.stringify({
+					username: newUserData.username,
+					countryCode: newUserData.countryCode,
+					company: newUserData.company,
+					mailListAgreed: newUserData.mailListAgreed,
+				}))}`),
+			);
+		});
+	});
+};
+
+const signupPost = () => {
+	describe('Sign Up', () => {
+		const newUserData = {
+			username: generateRandomString(),
+			countryCode: 'GB',
+			company: generateRandomString(),
+			mailListAgreed: true,
+		};
+
+		const newUserDataFromAad = {
+			mail: newUserEmail,
+			givenName: generateRandomString(),
+			surname: generateRandomString(),
+			id: generateRandomString(),
+		};
+
+		test('should fail if the email already exists', async () => {
+			const state = { ...newUserData };
+			Aad.getUserDetails.mockResolvedValueOnce({ data: { ...newUserDataFromAad, mail: userEmail } });
+			const res = await agent.get(`/v5/sso/aad/signup-post?state=${encodeURIComponent(JSON.stringify(state))}`)
+				.expect(templates.invalidArguments.status);
+			expect(res.body.code).toEqual(templates.invalidArguments.code);
+		});
+
+		test('should fail if the email already exists (SSO user)', async () => {
+			const state = { ...newUserData };
+			Aad.getUserDetails.mockResolvedValueOnce({ data: { ...newUserDataFromAad, mail: userEmailSso } });
+			const res = await agent.get(`/v5/sso/aad/signup-post?state=${encodeURIComponent(JSON.stringify(state))}`)
+				.expect(templates.invalidArguments.status);
+			expect(res.body.code).toEqual(templates.invalidArguments.code);
+		});
+
+		test('should sign a new user up', async () => {
+			const state = { ...newUserData };
+			Aad.getUserDetails.mockResolvedValueOnce({ data: newUserDataFromAad });
+			await agent.get(`/v5/sso/aad/signup-post?state=${encodeURIComponent(JSON.stringify(state))}`)
+				.expect(templates.ok.status);
```

Comment:
can we verify a user has been created? (like call getUserByUsername or something)